service/history/workflow/retry.go

```diff
@@ -131,3 +139,125 @@ func isRetryable(failure *failurepb.Failure, nonRetryableTypes []string) bool {
 	}
 	return true
 }
+
+// Helpers for creating new retry/cron workflows:
+
+// QUESTION: is this a good name? any suggestions?
+func SetupNewWorkflowForRetryOrCron(
+	oldMs MutableState,
+	newMs MutableState,
+	newRunId string,
+	startAttr *historypb.WorkflowExecutionStartedEventAttributes,
+	lastCompletionResult *commonpb.Payloads,
+	failure *failurepb.Failure,
+	backoffInterval time.Duration,
+	initiator enumspb.ContinueAsNewInitiator,
+) error {
+
+	// Extract ParentExecutionInfo from current run so it can be passed down to the next
+	var parentInfo *workflowspb.ParentExecutionInfo
+	executionInfo := oldMs.GetExecutionInfo()
+	if oldMs.HasParentExecution() {
+		parentInfo = &workflowspb.ParentExecutionInfo{
+			NamespaceId: executionInfo.ParentNamespaceId,
+			Namespace:   startAttr.GetParentWorkflowNamespace(),
+			Execution: &commonpb.WorkflowExecution{
+				WorkflowId: executionInfo.ParentWorkflowId,
+				RunId:      executionInfo.ParentRunId,
+			},
+			InitiatedId: executionInfo.InitiatedId,
+		}
+	}
+
+	newExecution := commonpb.WorkflowExecution{
+		WorkflowId: executionInfo.WorkflowId,
+		RunId:      newRunId,
+	}
+
+	firstRunID, err := oldMs.GetFirstRunID()
+	if err != nil {
+		return err
+	}
+
+	previousExecutionInfo := oldMs.GetExecutionInfo()
+	taskQueue := previousExecutionInfo.TaskQueue
+	if startAttr.TaskQueue != nil {
+		taskQueue = startAttr.TaskQueue.GetName()
+	}
+	tq := &taskqueuepb.TaskQueue{
+		Name: taskQueue,
+		Kind: enumspb.TASK_QUEUE_KIND_NORMAL,
+	}
+
+	workflowType := previousExecutionInfo.WorkflowTypeName
```

Comment:
can we unify the variable names? 
either use old vs new
or previous vs current

right now the variable names are not really clear